keep/api/tasks/process_event_task.py

```diff
@@ -368,16 +368,18 @@ def process_event(
         except Exception:
             logger.exception("Failed to run pre-formatting extraction rules")
 
-        if provider_type is not None:
+        if provider_type is not None and type(event) is dict:
             provider_class = ProvidersFactory.get_provider_class(provider_type)
             event = provider_class.format_alert(event, None)
 
+        # In case when provider_type is not set
+        if isinstance(event, dict):
+            event = [AlertDto(**event)]
```

Comment:
why did you move it?

Reply:
Do make it right after handling the dict with the provider. Looks better for me: Dict with provider -> dict without provider -> AlertDto...